app/models/user.rb

```diff
@@ -9,4 +9,16 @@ def to_s
   def self.authenticate(email, password)
     find_by(email: email).try(:authenticate, password)
   end
+
+  # def generate_token(column)
+  #   begin
+  #     self[column] = SecureRandom.urlsafe_base64
+  #   end while User.exists?(column: self[column])
+  # end
+
+  def send_password_reset
```

Comment:
this method name isn't quite truthful. you could move the mailer call in here, but then this becomes a little cluttered. I'd suggest renaming this to be more descriptive of it's contents. `set_password_reset_token` perhaps?